src/appleseed/renderer/kernel/rendering/progressive/progressiveframerenderer.cpp

```diff
@@ -500,8 +518,24 @@ namespace
                 if (m_abort_switch.is_aborted())
                     return;
 
+                // Compute current rendering time.
+                const double time = (m_timer.read() - m_start_time) * m_rcp_timer_freq;
+
+                // Compute sampling statistics.
+                const std::uint64_t samples = m_buffer.get_sample_count();
+                const double samples_per_pixel = samples * m_rcp_pixel_count;
+                m_sample_count_history_spinlock.lock();
```

Comment:
This does not affect progressive rendering perfs ?

Reply:
Nope, this code runs at the rate of 10 Hz (the default refresh rate) so essentially zero impact on performance.